src/porepy/models/fracture_damage.py

```diff
@@ -0,0 +1,326 @@
+
+        """
+        raise NotImplementedError("Subclass must implement this method.")
+
+
+class AnisotropicHistoryEquation(DamageHistoryEquation):
+    r"""Anisotropic damage history equation.
+
+    The anisotropic damage history equation is given by
+
+    .. math::
+
+        h = \int{\tau} max(m_t \cdot \delta u_t, 0) \, d\tau,
+
+    where :math:`\delta u_t` is the tangential plastic displacement jump increment and
+    :math:`m_t` is the normalized tangential plastic displacement jump, while h is the
+    damage history variable.
+
+    """
+
+    damage_history: Callable[[list[pp.Grid]], pp.ad.Variable]
+    """Damage history variable on ad form."""
+
+    plastic_displacement_jump: Callable[[list[pp.Grid]], pp.ad.Operator]
+    """Method returning plastic displacement jump."""
+
+    def damage_history_equation(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Anisotropic damage history equation.
+
+        Parameters:
+            subdomains: List of subdomains where the damage history is defined. Should
+                be of co-dimension one, i.e. fractures.
+
+            Returns:
+                Operator for the anisotropic damage history equation.
+
+        """
+        # Get the tangential component of the plastic displacement jump.
+        nd_vec_to_tangential = self.tangential_component(subdomains)
+        u_t: pp.ad.Operator = nd_vec_to_tangential @ self.plastic_displacement_jump(
+            subdomains
+        )
+        # The time increment of the tangential displacement jump.
+        u_t_increment: pp.ad.Operator = pp.ad.time_increment(u_t)
+        # Prepare for taking the inner product sum.
+        tangential_basis = self.basis(subdomains, dim=self.nd - 1)
+        tangential_to_scalar = pp.ad.sum_projection_list(
+            [e_i.T for e_i in tangential_basis]
+        )
+
+        m_t = self._normalized_tangential_plastic_jump(subdomains)
+        num_steps = self.time_manager.time_index
+        # The first term in the equation, the implicit part.
+        current_inner = tangential_to_scalar @ (m_t * u_t)
+        # We are only interested in positive values of the inner product.
+        f_heaviside = pp.ad.Function(partial(pp.ad.functions.heaviside, 0.5), "max")
+        f_abs = pp.ad.Function(pp.ad.functions.abs, "abs")
+        eq = self.damage_history(subdomains) - f_heaviside(current_inner) * f_abs(
+            tangential_to_scalar @ (m_t * u_t_increment)
+        )
+        # Then add the explicit part, i.e., the sum of the inner product of m with the u_t
+        # increment from all previous time steps. The sum starts at 1 since the first
+        # increment is already included in the implicit part.
+        for i in range(1, num_steps):
+            u_t_i = u_t.previous_timestep(i)
+            u_t_increment_i = u_t_i - u_t.previous_timestep(i + 1)
+            if np.allclose(
+                self.equation_system.evaluate(u_t_increment_i), 0, atol=1e-12
```

Comment:
I see the reasoning, and like the motivation, but I am not very happy with such assumptions on the absolute scaling of variables. I suppose you think it worth it, all things considered?

Reply:
Assuming yes, can we do a similar check on u_t_i? Or does my question above regarding u_t==0 come into play?

Reply:
Yes!